### PR TITLE
91 refactor and update documentation

### PR DESCRIPTION
**Pull request by ASimpleFox**
Refactored and updated the API Documentation, split the whole documentation into four manageable chunks. Mainly just for readability, since we're actually getting into connecting back-end and front-end!

Change log:
- Deleted original API Documentation file and refactored it into 4 smaller files
- 4 smaller sub-files including a generalized Overview file (for calling endpoints via Heroku), and three other files for the main functionalities of the API
- Added sample calls for urls with required parameters
- Fixed some documentation in the `/get` endpoint

Notes:
- Sample calls were not added for urls with body parameters. This is because body parameters are not encoded into the actual URL, rather are sent in the Request Body, so it's rather hard to write sample calls for it.

### CODE REVIEW

**Review by nith2001 — DISMISSED** [review]
Looks solid! Just a spelling mistake on "occurred" in the Overview.md, easy fix

File: back-end/API_Documentation/Overview.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+* Example: Get books from bookshelf = `https://husky-reads.herokuapp.com/bookshelves/get/elliot/reading`
+
+## <span style="color:deepskyblue">Server Errors</span>
+---
+All server errors will return the following content:
+
+If text response:
+
+  * **Code:** 500 </br>
+    **Content:** `"An error has occured on the server"`
```
Comment (nith2001) [inline back-end/API_Documentation/Overview.md:26]: Right here

Comment (ASimpleFox) [reply back-end/API_Documentation/Overview.md]: Typos updated

**Review by nith2001 — DISMISSED** [review]
Cool!

File: back-end/API_Documentation/AuthAndUser.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+### Grabbing username from cookie
+
+Grabs the username of the user that was previously logged in. Note that username cookies expire after 2 days.
+* **Endpoint:** /grab/username
+
+* **Request Method:** GET
+
+* **Returned Data Format:** JSON
+
+* **Sample Request: `/grab/elliot`**
```
Comment (ElliotSchumacher) [inline back-end/API_Documentation/AuthAndUser.md:167]: The "username" in the endpoint `/grab/username` is just part of the path to access that given endpoint. It is not a parameter and no sample request is needed. In order to use this endpoint, the front-end just needs to visit the `/grab/username` endpoint with a cookie specifying a user. **We should delete this sample request**

**Review by ElliotSchumacher — CHANGES_REQUESTED** [review]
The refactoring looks great and I think it will be much easier for the audience to find the specific endpoint they are working on. I found one mistake regarding a sample request for one of the `grab` endpoints. I don't think it is necessary and the sample call is written incorrectly.

File: back-end/API_Documentation/AuthAndUser.md (hunk 2 above)
Comment (ASimpleFox) [reply back-end/API_Documentation/AuthAndUser.md]: Patched, please review

**Review by nith2001 — APPROVED** [review]
Looks good!

**Pull request merged**